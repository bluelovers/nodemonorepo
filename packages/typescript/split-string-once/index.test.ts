import splitOnce from './index'

it('matches snapshot', () => {
  const mkfn = (separator: string) => (text: string) => ({
    input: {
      text,
      separator
    },
    output: splitOnce(text, separator)
  })

  expect([
    [
      'abc,def,ghi,jkl',
      'a,bc,def,ghij',
      ',abc,def,ghi,jkl',
      'abc,def,ghi,jkl,',
      'abcdefghijkl'
    ].map(mkfn(',')),
    [
      'abc//def/ghi//jkl///mno',
      'abc//def//ghi//jkl'
    ].map(mkfn('//'))
  ]).toMatchSnapshot()
})
