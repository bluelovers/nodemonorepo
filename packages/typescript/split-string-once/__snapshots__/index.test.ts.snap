// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  Array [
    Object {
      "input": Object {
        "separator": ",",
        "text": "abc,def,ghi,jkl",
      },
      "output": Array [
        "abc",
        "def,ghi,jkl",
      ],
    },
    Object {
      "input": Object {
        "separator": ",",
        "text": "a,bc,def,ghij",
      },
      "output": Array [
        "a",
        "bc,def,ghij",
      ],
    },
    Object {
      "input": Object {
        "separator": ",",
        "text": ",abc,def,ghi,jkl",
      },
      "output": Array [
        "",
        "abc,def,ghi,jkl",
      ],
    },
    Object {
      "input": Object {
        "separator": ",",
        "text": "abc,def,ghi,jkl,",
      },
      "output": Array [
        "abc",
        "def,ghi,jkl,",
      ],
    },
    Object {
      "input": Object {
        "separator": ",",
        "text": "abcdefghijkl",
      },
      "output": Array [
        "abcdefghijkl",
      ],
    },
  ],
  Array [
    Object {
      "input": Object {
        "separator": "//",
        "text": "abc//def/ghi//jkl///mno",
      },
      "output": Array [
        "abc",
        "def/ghi//jkl///mno",
      ],
    },
    Object {
      "input": Object {
        "separator": "//",
        "text": "abc//def//ghi//jkl",
      },
      "output": Array [
        "abc",
        "def//ghi//jkl",
      ],
    },
  ],
]
`;
